Std/Data/Array/Lemmas.lean

```diff
@@ -108,7 +106,7 @@ theorem get?_push {a : Array α} : (a.push x)[i]? = if i = a.size then some x el
 @[simp] theorem data_set (a : Array α) (i v) : (a.set i v).data = a.data.set i.1 v := rfl
 
 theorem get_set_eq (a : Array α) (i : Fin a.size) (v : α) :
-    (a.set i v)[i.1]'(by simp [i.2]) = v := by
+    (a.set i v)[i.1] = v := by
   simp only [set, getElem_eq_data_get, List.get_set_eq]
```

Comment:
It feels like `get_set_eq` should replace the `(i : Fin a.size) with `i : Nat` and a separate proof.  The current proof seems overly specific.

Reply:
This may not work with rewriting if the argument to `set` is not a `Fin.mk` application.

Reply:
I don't follow but don't need to.  If these changes are useful to others, I am fine with merging them.